x/delayedack/keeper/rollapp_packets_list_filter.go

```diff
@@ -0,0 +1,58 @@
+package keeper
+
+import commontypes "github.com/dymensionxyz/dymension/v3/x/common/types"
+
+type filterFunc func(val commontypes.RollappPacket) bool
+
+type rollappPacketListFilter struct {
+	prefixes        [][]byte
+	filter          filterFunc
+	breakOnMismatch bool
+}
+
+func AllRollappPackets() rollappPacketListFilter {
+	return rollappPacketListFilter{prefixes: [][]byte{{}}}
+}
+
+func ByRollappIDAndStatus(rollappID string, status ...commontypes.Status) rollappPacketListFilter {
+	return rollappPacketListFilter{
+		prefixes: buildPrefixes(rollappID, status),
+	}
+}
+func ByRollappIDAndStatusAndMaxHeight(
+	rollappID string,
+	maxProofHeight uint64,
+	breakOnMismatch bool,
+	status ...commontypes.Status,
+) rollappPacketListFilter {
+	filter := ByRollappIDAndStatus(rollappID, status...)
+	filter.breakOnMismatch = breakOnMismatch
+	filter.filter = func(val commontypes.RollappPacket) bool {
```

Comment:
why u need filter for it?
can't u use `status/rollappID/proofHeight` for proof height iterator?
otherwise u get all the heights and filtering afterwards

Reply:
i believe you can't as `status/rollappID/proofHeight` will give you all packets at height and you need all packets until height.

Reply:
if `proofHeight` is ordered, should be possible to iterate `until height`

Reply:
> otherwise u get all the heights and filtering afterwards

I believe the iterator gets a batch from the db every time (and not all elements),  and it does so only when  `next()` element is the last in that batch. 

in that sense, you'll make the same amount of db calls assuming it gets the height by order, which I believe it does (as you're breaking from the iterator when you reached max height). 

If that's the case, than the only mis-optimization I see here is holding more objects in memory than necessary (i.e the last batch could potentially contain less objects if the height was limited)

Not sure it's worth the effort of changing,  optimization wise.